### PR TITLE
fix: use kitty user-vars for for determining if active window is running neovim

### PR DESCRIPTION
**Pull request by oddnugget**
Attempts fixing: #147

I'm not well versed in Python,  but i have this working in my setup so i thought i would contribute. Not sure about the best  place to call the user-var kitten on the nvim side.

### CODE REVIEW

**Comment by kovidgoyal** [comment]
Note that you dont need to use remote control, or indeed even a kitten for this. You can use the new-ish conditional mappings feature in kitty.
https://sw.kovidgoyal.net/kitty/mapping/#conditional-mappings-depending-on-the-state-of-the-focused-window

That shows you how to configure vim and neovim to set a user var in kitty when they are active.

**Comment by oddnugget** [comment]
Oh wow! That was actually a lot smoother. Thanks! I must have missed it in the documentation. I'll try and update the PR when i get some time

**Comment by mrjones2014** [comment]
Note that if we update to remove the `pass_keys.py` kitten in favor of the new conditional mappings in Kitty, we will also need to remove this line from `kitty/install_kittens.bash`

https://github.com/mrjones2014/smart-splits.nvim/blob/50f52146e4504a3fc0f0d5830c8560a16a95dd08/kitty/install-kittens.bash#L5

**Comment by oddnugget** [comment]
> Note that if we update to remove the `pass_keys.py` kitten in favor of the new conditional mappings in Kitty, we will also need to remove this line from `kitty/install_kittens.bash`
> 
> https://github.com/mrjones2014/smart-splits.nvim/blob/50f52146e4504a3fc0f0d5830c8560a16a95dd08/kitty/install-kittens.bash#L5

it seems relative_resize_window is duplicated between `pass-keys.py` and `relative-resize.py`. Should i go ahead and delete pass-keys altogether?

**Review by mrjones2014 — APPROVED** [review]

**Comment by mrjones2014** [comment]
> it seems relative_resize_window is duplicated between pass-keys.py and relative-resize.py. Should i go ahead and delete pass-keys altogether?

@oddnugget so sorry this totally fell off my radar, apologies! Yeah, if I understand correctly, using Kitty user vars means we don't needs `pass_keys.py` anymore, since all that did was determine whether to consume the keys at the Kitty level or to pass them on to Neovim, which is now handled by Kitty conditional mappings.

**Comment by oddnugget** [comment]
Totally forgot about this one. Will have a look later today

File: lua/smart-splits/mux/kitty.lua [hunk 1]
```diff
@@ -97,6 +97,22 @@ function M.resize_pane(direction, amount)
   return ok
 end
 
+function M.on_init()
+  vim.api.nvim_create_autocmd('VimEnter', {
+    group = vim.api.nvim_create_augroup('KittySetVarVimEnter', { clear = true }),
+    callback = function()
+      io.stdout:write('\x1b]1337;SetUserVar=in_editor=MQo\007')
```
[the remark this answers is not in the document]
Comment (oddnugget) [reply lua/smart-splits/mux/kitty.lua]: Sure!

File: lua/smart-splits/mux/kitty.lua [hunk 2]
```diff
@@ -97,6 +97,22 @@ function M.resize_pane(direction, amount)
...
+      io.stdout:write('\x1b]1337;SetUserVar=IS_NVIM=MQo\007')
+    end,
+  })
+
+  vim.api.nvim_create_autocmd('VimLeave', {
+    group = vim.api.nvim_create_augroup('KittyUnsetVarVimLeave', { clear = true }),
+    callback = function()
+      io.stdout:write('\x1b]1337;SetUserVar=IS_NVIM\007')
+    end,
+  })
```
Comment (mrjones2014) [inline lua/smart-splits/mux/kitty.lua:113]: One other thing I forgot to mention; you shouldn't need to manually create the autocmds here. The way the API works is, the mux backend should provide an `on_init` and an `on_exit`, and the shared startup procedure handles creating the autocmds. This way `VimSuspend` and `VimResume` are handled automatically as well.

https://github.com/mrjones2014/smart-splits.nvim/blob/95d11bfc83263236431ead7dc017f5ab0ff992ca/lua/smart-splits/mux/utils.lua#L46-L78

So basically, you should just have:

```lua
function M.on_init()
  io.stdout:write('\x1b]1337;SetUserVar=IS_NVIM=MQo\007')
end

function M.on_exit()
  io.stdout:write('\x1b]1337;SetUserVar=IS_NVIM\007')
end
```

**Review by mrjones2014 — CHANGES_REQUESTED** [review]
We just need to tweak it a bit to adhere to the mux startup API as noted above.

File: lua/smart-splits/mux/kitty.lua (hunk 2 above)
Comment (oddnugget) [reply lua/smart-splits/mux/kitty.lua]: Ah! I sort of hastily added them without really exploring the rest of the code. Fixed now!

File: doc/smart-splits.txt [hunk 3]
```diff
@@ -484,9 +484,6 @@ KITTY                                                     *smart-splits-kitty*
   [!NOTE] The `config.at_edge = 'wrap'` option is not supoprted in Kitty
   terminal multiplexer due to inability to determine pane layout from CLI.
 
-  [!NOTE] This won't work if the pane is connected over SSH, as the pane will
```
Comment (mrjones2014) [inline doc/smart-splits.txt:487]: This file is actually auto-generated; can you update the README.md instead?

**Review by mrjones2014 — CHANGES_REQUESTED** [review]
Sorry I didn't notice this before! This is the last change needed, I promise!

**Comment by oddnugget** [comment]
Update the README. I haven't tested the relative resizing, don't use that feature, but it should work the same way i presume.

**Review by mrjones2014 — APPROVED** [review]
Thanks so much for the PR! This is a great improvement.

**Comment by oddnugget** [comment]
Happy to help :)
